Automattic/jetpack
# Extensions: Add extensions building

- Adds block building to Jetpack — `yarn build-extensions`
- Update tooling (pre-commit, eslint, reformat-files) for extensions

More fun? Watch and build! `yarn build-extensions --watch` 💥 

## Blockers
- [x] #11677 (fix CommonJS / esmodules imports)
- [x] #10810 (Upgrade babel for existing React app: Jetpack dashboard/settings)

## Testing

- `yarn` (update/install deps)
- Extensions build as expected and work correctly:
  - Clean (`yarn clean-client`) -- nothing in `_inc/blocks`
  - Build (`yarn build-extensions`) Built and working!
- Check a few things in the output:
  - Output is what we expect:
    - no references to React or ReactDOM
    - es6 is properly transpiled (e.g. no plain `class` declarations)
  - Production builds are minified (`NODE_ENV=production yarn build-extensions`)
- Run a full build and check for regressions (`yarn clean-client && yarn build`)
  - Do the blocks and extensions continue to work well?
  - React-based Dashboard works well?

## Changelog

- Bring the Block Editor extensions source and build tooling into the Jetpack repo

## Concerns

After #10810 lands, none 😎 

~Two versions of babel and babel deps~

~The React based Jetpack dashboard depends on older versions of `babel`. I suspect it's find to use both because these versions happen not to collide (`babel-thing@6` vs `@babel/thing@7`)~
